onnxruntime/core/providers/webnn/builders/impl/gemm_op_builder.cc

```diff
@@ -38,16 +38,7 @@ Status GemmOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
   emscripten::val b = model_builder.GetOperand(node.InputDefs()[b_idx]->Name());
   emscripten::val output = emscripten::val::object();
   if (op_type == "MatMul") {
-    std::vector<int64_t> a_shape;
-    if (!GetShape(*input_defs[a_idx], a_shape, logger)) {
-      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Can not get shape of A.");
-    }
-    // The inputs of matmul must be at least 3D for WebNN CPU EP, use gemm for 2D case.
-    if (model_builder.GetWebnnDeviceType() == WebnnDeviceType::CPU && a_shape.size() == 2) {
-      output = model_builder.GetBuilder().call<emscripten::val>("gemm", a, b);
-    } else {
-      output = model_builder.GetBuilder().call<emscripten::val>("matmul", a, b);
-    }
```

Comment:
@zesongw , please bring this change back and add a todo. We need this workaround until it is fixed in Chromium XNNPACK backend.